stac_generator/csv/utils.py

```diff
@@ -0,0 +1,248 @@
+    if columns:
+        if isinstance(columns[0], str):
+            usecols = list(columns)
+        else:
+            usecols = [item["name"] for item in cast(list[ColumnInfo], columns)]
+        usecols.extend([X_coord, Y_coord])
+        if T_coord:
+            usecols.append(T_coord)
+        # If item group provided -> read in
+        if groupby:
+            usecols.extend(groupby)
+    return cast(
+        PDFrameT,
+        pd.read_csv(src_path, parse_dates=parse_dates, date_format=date_format, usecols=usecols),  # type: ignore[call-overload]
+    )
+
+
+def to_gdf(df: PDFrameT, X_coord: str, Y_coord: str, epsg: int) -> FrameT:
+    return gpd.GeoDataFrame(df, geometry=gpd.points_from_xy(df[X_coord], df[Y_coord], crs=epsg))
+
+
+def calculate_temporal_extent(
+    df: FrameT | None = None,
+    time_col: str | None = None,
+    datetime: DateTimeT | None = None,
+    start_datetime: DateTimeT | None = None,
+    end_datetime: DateTimeT | None = None,
+) -> TimeExtentT:
+    """Get temporal extent based on Stac specification.
+
+    Use `start_datetime` and `end_datetime` if provided. Otherwise, use `datetime` if provided.
+    If the dataframe and time column are provided, the function will obtain the start datetime and end datetime from the dataframe.
+
+    :param df: Provided dataframe.
+    :type df: FrameT | None, optional
+
+    :param time_col: Name of time column.
+    :type time_col: str | None, optional
+
+    :param datetime: Datetime.
+    :type datetime: DateTimeT | None, optional
+
+    :param start_datetime: Start datetime.
+    :type start_datetime: DateTimeT | None, optional
+
+    :param end_datetime: End datetime.
+    :type end_datetime: DateTimeT | None, optional
+
+    :return: Start datetime, end datetime.
+    :rtype: TimeExtentT
+    """
+    if start_datetime and end_datetime:
+        return start_datetime, end_datetime
+    if datetime:
+        return datetime, datetime
+    if df is not None and isinstance(time_col, str):
+        if time_col not in df.columns:
+            raise KeyError(f"Cannot find time_col: {time_col} in given dataframe")
+        if not isinstance(df[time_col].dtype, DateTimeT):
+            raise ValueError(
+                f"Dtype of time_col: {time_col} must be of datetime type: {df[time_col].dtype}"
+            )
+        min_T, max_T = df[time_col].min(), df[time_col].max()
+        return (min_T, max_T)
+    raise ValueError(
+        "If datetime is None, both start_datetime and end_datetime values must be provided"
+    )
+
+
+def calculate_geometry(
+    df: FrameT,
+) -> Point | MultiPoint:
+    """Calculate the geometry from geopandas dataframe.
+
+    Returns a `shapely.Point` or `shapely.MultiPoint` depending on the number
+    of unique points in the dataframe.
+
+    :param df: source dataframe
+    :type df: FrameT
+    :return: shapely geometry object
+    :rtype: Point | MultiPoint
+    """
+    points: Sequence[Point] = df["geometry"].unique()
```

Comment:
geoDataFrame can have polygons or other Geometry types in the geometry column. We can probably assume that we are taking only points when reading from csv but maybe we can explicit that this only works for this case.